app/models/container.rb

```diff
@@ -57,12 +57,9 @@ class Container < ApplicationRecord
   #
   # Setup additional methods
   #
-  def self.create_with_source(cluster_id, params)
-    container = Container.new
-    unless params[:source].present?
-      container.errors.add(:source, "must be present.")
-      return container
-    end
+  def apply_with_source(params)
```

Comment:
what does `apply` means?

Reply:
`apply` in this context means putting the parameters into the current container, not creating it.

It's sort of `assign_attributes`.

Reply:
cool, since we are not using [type annotation](https://sorbet.org/) here, I would suggest the name be more verbose.. e.g. `apply_params_with_source`

Reply:
Noted.